--- a/libglycin/meson.build
+++ b/libglycin/meson.build
@@ -5,7 +5,7 @@
 libglycin_deps = [
     dependency('fontconfig', version: fontconfig_req),
     dependency('gio-2.0', version: gio_req),
-    dependency('libseccomp', version: seccomp_req),
+    dependency('libseccomp', version: seccomp_req, required: host_machine.system() == 'linux'),
     dependency('lcms2', version: lcms2_req),
 ]

@@ -53,6 +53,16 @@

 global_libglycin_gir = 'non-existent'

+cc = meson.get_compiler('c')
+if host_machine.system() == 'darwin'
+    lib_ext = 'dylib'
+elif host_machine.system() == 'windows'
+    lib_ext = 'dll'
+else
+    lib_ext = 'so'
+endif
+lib_prefix = cc.get_argument_syntax() == 'msvc' ? '' : 'lib'
+
 foreach package : packages
     suffix = package['suffix']
     namespace_suffix = package['namespace_suffix']
@@ -72,7 +82,7 @@
         endif
     endif

-    cargo_output = cargo_target_dir / rust_target / f'libglycin@suffix@.so'.replace('-', '_')
+    cargo_output = cargo_target_dir / rust_target / (lib_prefix + f'glycin@suffix@'.replace('-', '_') + '.' + lib_ext)

     # Build library with Rust
     libglycin = custom_target(
@@ -109,5 +119,6 @@
         link_with: link_with,
         include_directories: 'include/',
         dependencies: deps,
+        link_args: cc.get_argument_syntax() == 'msvc' ? ['/NOENTRY'] : [],
         # Mark dirty for "generate_gir" if the header or the rust generated .so changes
         link_depends: [
--- a/libglycin/build.rs
+++ b/libglycin/build.rs
@@ -1,5 +1,8 @@
 fn main() {
     let major = std::env::var("CARGO_PKG_VERSION_MAJOR").unwrap();
-    // Set soname of library
-    println!("cargo:rustc-cdylib-link-arg=-Wl,-soname,libglycin-{major}.so.0");
+    match std::env::var("CARGO_CFG_TARGET_OS").unwrap().as_str() {
+        "linux" => println!("cargo:rustc-cdylib-link-arg=-Wl,-soname,libglycin-{major}.so.0"),
+        "macos" => println!("cargo:rustc-cdylib-link-arg=-Wl,-install_name,@rpath/libglycin-{major}.dylib"),
+        _ => {}
+    }
 }
--- a/libglycin/libglycin-gtk4/build.rs
+++ b/libglycin/libglycin-gtk4/build.rs
@@ -1,7 +1,10 @@
 fn main() {
     let major = std::env::var("CARGO_PKG_VERSION_MAJOR").unwrap();
-    // Set soname of library
-    println!("cargo:rustc-cdylib-link-arg=-Wl,-soname,libglycin-gtk4-{major}.so.0");
+    match std::env::var("CARGO_CFG_TARGET_OS").unwrap().as_str() {
+        "linux" => println!("cargo:rustc-cdylib-link-arg=-Wl,-soname,libglycin-gtk4-{major}.so.0"),
+        "macos" => println!("cargo:rustc-cdylib-link-arg=-Wl,-install_name,@rpath/libglycin-gtk4-{major}.dylib"),
+        _ => {}
+    }

     system_deps::Config::new().probe().unwrap();
 }
