diff --git a/libglycin/build.rs b/libglycin/build.rs
index 4b2c5ee..24fc443 100644
--- a/libglycin/build.rs
+++ b/libglycin/build.rs
@@ -1,5 +1,10 @@
 fn main() {
     let major = std::env::var("CARGO_PKG_VERSION_MAJOR").unwrap();
+
     // Set soname of library
-    println!("cargo:rustc-cdylib-link-arg=-Wl,-soname,libglycin-{major}.so.0");
+    match std::env::var("CARGO_CFG_TARGET_OS").unwrap().as_str() {
+        "linux" => println!("cargo:rustc-cdylib-link-arg=-Wl,-soname,libglycin-{major}.so.0"),
+        "macos" => println!("cargo:rustc-cdylib-link-arg=-Wl,-install_name,@rpath/libglycin-{major}.0.dylib"),
+        _ => {}
+    }
 }
diff --git a/libglycin/libglycin-gtk4/build.rs b/libglycin/libglycin-gtk4/build.rs
index fee26cc..2a01b89 100644
--- a/libglycin/libglycin-gtk4/build.rs
+++ b/libglycin/libglycin-gtk4/build.rs
@@ -1,7 +1,12 @@
 fn main() {
     let major = std::env::var("CARGO_PKG_VERSION_MAJOR").unwrap();
+
     // Set soname of library
-    println!("cargo:rustc-cdylib-link-arg=-Wl,-soname,libglycin-gtk4-{major}.so.0");
+    match std::env::var("CARGO_CFG_TARGET_OS").unwrap().as_str() {
+        "linux" => println!("cargo:rustc-cdylib-link-arg=-Wl,-soname,libglycin-gtk4-{major}.so.0"),
+        "macos" => println!("cargo:rustc-cdylib-link-arg=-Wl,-install_name,@rpath/libglycin-gtk4-{major}.0.dylib"),
+        _ => {}
+    }
 
     system_deps::Config::new().probe().unwrap();
 }
diff --git a/libglycin/meson.build b/libglycin/meson.build
index 12cbd5f..e30bc81 100644
--- a/libglycin/meson.build
+++ b/libglycin/meson.build
@@ -5,7 +5,7 @@ endif
 libglycin_deps = [
     dependency('fontconfig', version: fontconfig_req),
     dependency('gio-2.0', version: gio_req),
-    dependency('libseccomp', version: seccomp_req),
+    dependency('libseccomp', version: seccomp_req, required: host_machine.system() == 'linux'),
     dependency('lcms2', version: lcms2_req),
 ]
 
@@ -53,6 +53,16 @@ endif
 
 global_libglycin_gir = 'non-existent'
 
+cc = meson.get_compiler('c')
+if host_machine.system() == 'darwin'
+    lib_ext = 'dylib'
+elif host_machine.system() == 'windows'
+    lib_ext = 'dll'
+else
+    lib_ext = 'so'
+endif
+lib_prefix = cc.get_argument_syntax() == 'msvc' ? '' : 'lib'
+
 foreach package : packages
     suffix = package['suffix']
     namespace_suffix = package['namespace_suffix']
@@ -72,7 +82,8 @@ foreach package : packages
         endif
     endif
 
-    cargo_output = cargo_target_dir / rust_target / f'libglycin@suffix@.so'.replace('-', '_')
+    crate_path = suffix == '' ? 'libglycin' : f'libglycin/libglycin@suffix@'
+    cargo_output = cargo_target_dir / rust_target / (lib_prefix + f'glycin@suffix@'.replace('-', '_') + '.' + lib_ext)
 
     # Build library with Rust
     libglycin = custom_target(
@@ -109,6 +120,7 @@ foreach package : packages
         link_with: link_with,
         include_directories: 'include/',
         dependencies: deps,
+        link_args: cc.get_argument_syntax() == 'msvc' ? ['/NOENTRY'] : [],
         # Mark dirty for "generate_gir" if the header or the rust generated .so changes
         link_depends: [
             # For unknown reasons the header triggers repeated gir regenerates until something else changes
@@ -148,16 +160,7 @@ foreach package : packages
         sources: libglycin_copy_library,
     )
 
-    if suffix == ''
-        version = run_command(crate_version_bin, f'libglycin@suffix@', 'gnome', check: true).stdout()
-    else
-        version = run_command(
-            crate_version_bin,
-            f'libglycin/libglycin@suffix@',
-            'gnome',
-            check: true,
-        ).stdout()
-    endif
+    version = run_command(crate_version_bin, crate_path, 'gnome', check: true).stdout()
 
     # Generate .pc file
     pkg.generate(
